pkg/sql/schemachanger/scdeps/build_deps.go

```diff
@@ -318,3 +380,18 @@ func (d *buildDeps) IncrementUserDefinedSchemaCounter(
 func (d *buildDeps) IncrementEnumCounter(counterType sqltelemetry.EnumTelemetryType) {
 	sqltelemetry.IncrementEnumCounter(counterType)
 }
+
+// currentSchema returns the first schema in the schema search path.
+func (d *buildDeps) currentSchema(ctx context.Context) string {
+	iter := d.schemaResolver.CurrentSearchPath().IterWithoutImplicitPGSchemas()
+	for schemaName, ok := iter.Next(); ok; schemaName, ok = iter.Next() {
+		found, scMeta, err := d.schemaResolver.LookupSchema(ctx, d.CurrentDatabase(), schemaName)
+		if err != nil {
+			panic(err)
+		}
+		if found {
+			return scMeta.Schema.GetName()
+		}
+	}
+	panic("There is not a single valid schema in search path.")
```

Comment:
You need to panic errors in this layer. Non-errors will be turned into assertion failures.